### PR TITLE
rootdir: Android.mk: Create flags instead of filter

### PR DESCRIPTION
**Pull request by Haxk20**
Instead of adding each SoC that needs some module
make the module behind a flag which can be included
in the device tree for the platform.

### CODE REVIEW

**Review by MarijnS95 — APPROVED** [review]

File: rootdir/Android.mk [hunk 1]
```diff
@@ -100,27 +100,27 @@ LOCAL_SRC_FILES := vendor/etc/init/cdsprpcd.rc
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR)/etc/init
 include $(BUILD_PREBUILT)
-endif # TARGET_BOARD_PLATFORM in (sdm845 sm6125 sm6350 sm8150)
+endif
 
-ifneq ($(filter sdm660 msm8998 sdm845 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifeq ($(TARGET_NEEDS_SSCRPCD), true)
```
Comment (MarijnS95) [inline rootdir/Android.mk:105]: As mentioned in Kumano, I think we already have `TARGET_USES_SSC` for this.

File: rootdir/Android.mk [hunk 2]
```diff
@@ -83,7 +83,7 @@ LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR)/etc/init
 include $(BUILD_PREBUILT)
 endif
 
-ifneq ($(filter sdm845 sm6125 sm6350 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifeq ($(TARGET_HAS_CDSP), true)
```
Comment (MarijnS95) [inline rootdir/Android.mk:86]: Following SSC, `TARGET_USES_CDSP`?

File: rootdir/Android.mk [hunk 3]
```diff
@@ -100,27 +100,27 @@ LOCAL_SRC_FILES := vendor/etc/init/cdsprpcd.rc
...
 LOCAL_MODULE_CLASS := ETC
 LOCAL_SRC_FILES := vendor/etc/init/sscrpcd.rc
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR)/etc/init
 include $(BUILD_PREBUILT)
-endif # TARGET_BOARD_PLATFORM in (sdm660 msm8998 sdm845 sm8150)
+endif
 
-ifneq ($(filter sm6125 sm6350 ,$(TARGET_BOARD_PLATFORM)),)
+ifeq ($(TARGET_NEEDS_HVDCP_OPTI), true)
```
Comment (MarijnS95) [inline rootdir/Android.mk:115]: Can we make this one more explanatory? `TARGET_NEEDS_USERSPACE_CHARGER_CURVE_GARBAGE_COMPUTERATOR`?

Comment (Haxk20) [reply rootdir/Android.mk:115]: No we cant.

Comment (MarijnS95) [reply rootdir/Android.mk:115]: If I want it, I can :P

**Pull request closed without merging**